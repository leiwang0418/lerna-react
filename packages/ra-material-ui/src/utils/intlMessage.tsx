const intlMessage: any = (locale: string) => {
	switch (locale) {
		case "en":
			return import("../compiled-lang/zh.json");
		default:
			return import("../compiled-lang/zh.json");
	}
};

export default intlMessage;
