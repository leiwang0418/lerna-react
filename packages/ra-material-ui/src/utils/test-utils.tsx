import React, { FC } from "react";
import { render, RenderOptions } from "@testing-library/react";

// const customRender = (
// 	ui: React.ReactElement,
// 	{...renderOptions } = {}
// ) => {
// 	const Wrapper: FC = ({ children }) => (
// 		<IntlProvider locale={locale} messages={zh}>
// 			<Provider store={store}>
// 				<Router>{children}</Router>
// 			</Provider>
// 		</IntlProvider>
// 	);

// 	return render(ui, { wrapper: Wrapper, ...renderOptions });
// };
const customRender = render;

export * from "@testing-library/react";

export { customRender as render };
