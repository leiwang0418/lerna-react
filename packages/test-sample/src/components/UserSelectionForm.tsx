import React, { FC, useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl, defineMessage } from 'react-intl';

import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import Button from '@material-ui/core/Button';
import AccountCircle from '@material-ui/icons/AccountCircle';
import { makeStyles } from '@material-ui/core/styles';

const GitHubUserInputProps = {
	startAdornment: (
		<InputAdornment position="start">
			<AccountCircle />
		</InputAdornment>
	),
};

interface Props {
	username?: string;
	setUsername: Function;
}

const useStyles = makeStyles((theme) => ({
	button: {
		marginTop: theme.spacing(1),
	},
}));

const message = {
	searchFiled: {
		label: defineMessage({
			defaultMessage: 'GitHub用户名',
			description: '用户名查询框label',
		}),
		placeholder: defineMessage({
			defaultMessage: '请输入GitHub用户名',
			description: '用户名查询框label默认提示',
		}),
	},
	submitText: defineMessage({
		defaultMessage: '查看仓库列表',
		description: '查看仓库列表提交按钮显示内容',
	}),
};

const UserSelectionForm: FC<Props> = ({ username, setUsername }) => {
	const intl = useIntl();
	const classes = useStyles();
	const [newUsername, setNewUsername] = useState(username);

	const handleSubmit = (e: React.SyntheticEvent) => {
		e.preventDefault();
		setUsername(newUsername);
	};

	const handleChangeUsername: React.ChangeEventHandler<HTMLInputElement> = (
		e
	) => {
		const value = e.currentTarget.value;
		setNewUsername(value);
	};

	return (
		<form onSubmit={handleSubmit}>
			<Grid container alignItems="center">
				<Grid item xs={12}>
					<TextField
						label={intl.formatMessage(message.searchFiled.label)}
						placeholder={intl.formatMessage(
							message.searchFiled.placeholder
						)}
						InputProps={GitHubUserInputProps}
						required
						fullWidth
						value={newUsername}
						onChange={handleChangeUsername}
					/>
				</Grid>
				<Grid item xs={12}>
					<Button
						variant="contained"
						color="primary"
						type="submit"
						fullWidth
						className={classes.button}
					>
						{intl.formatMessage(message.searchFiled.submitText)}
					</Button>
				</Grid>
			</Grid>
		</form>
	);
};

UserSelectionForm.defaultProps = {
	username: 'leiwang0418',
};

UserSelectionForm.propTypes = {
	username: PropTypes.string,
	setUsername: PropTypes.func.isRequired,
};

export default UserSelectionForm;
