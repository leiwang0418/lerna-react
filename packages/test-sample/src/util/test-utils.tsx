import React, { FC } from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import zh from '../compiled-lang/zh.json';

const render = (
	ui: React.ReactElement,
	{ locale = 'zh', ...renderOptions } = {}
) => {
	const Wrapper: FC = ({ children }) => (
		<IntlProvider locale={locale} messages={zh}>
			{children}
		</IntlProvider>
	);

	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from '@testing-library/react';

export { render };
