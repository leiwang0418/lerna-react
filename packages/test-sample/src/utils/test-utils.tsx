import React, { FC } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import zh from "../compiled-lang/zh.json";
import { Provider } from "react-redux";
import { rootReducer } from "../store";

import { BrowserRouter as Router } from "react-router-dom";
import { createStore, combineReducers, applyMiddleware } from "redux";
import thunk from "redux-thunk";

export const configStore = () => {
	return createStore(combineReducers(rootReducer), applyMiddleware(thunk));
};

const customRender = (
	ui: React.ReactElement,
	{ locale = "zh", store = configStore(), ...renderOptions } = {}
) => {
	const Wrapper: FC = ({ children }) => (
		<IntlProvider locale={locale} messages={zh}>
			<Provider store={store}>
				<Router>{children}</Router>
			</Provider>
		</IntlProvider>
	);

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from "@testing-library/react";

export { customRender as render };
export const store = configStore();
