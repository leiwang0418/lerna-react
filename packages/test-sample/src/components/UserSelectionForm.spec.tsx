import React from 'react';
import { render, screen, fireEvent } from '../utils/test-utils';
import UserSelectionForm from './UserSelectionForm';


import { Provider, useDispatch } from 'react-redux';
import configureStore from 'redux-mock-store';


const setup = () => {
	const mockSetUsername = jest.spyOn(require("../pages/HomeSlice"), 'setUsername');
	const initialState = { user: { username: 'leiwang0418', isEditing: true } };
	const mockStore = configureStore();
	const store = mockStore(initialState);

	const { container } = render(
		<Provider store={store}>
			<UserSelectionForm />
		</Provider>
	);

	const input = screen.getByDisplayValue(/leiwang0418/i);
	const submit = screen.getByRole('button');

	return { input, submit, mockSetUsername, container };
};

describe('snapshot test', () => {
	it('renders the basic UserSelectionForm', () => {
		const { container } = setup();

		expect(container).toMatchSnapshot();

		expect(screen.getByPlaceholderText(/请输入GitHub用户名/i)).toBeInTheDocument();
		expect(screen.getByDisplayValue(/leiwang0418/i)).toBeInTheDocument();
		expect(screen.getByRole('button')).toHaveTextContent('查看仓库列表');
	});
});

describe('act test', () => {
	it('change search value', () => {
		const { input } = setup();
		fireEvent.change(input, { target: { value: 'lei' } });

		expect(screen.getByDisplayValue('lei') === input).toBe(true);
	});

	it('submit clicked', () => {
		const { mockSetUsername, input, submit } = setup();
		fireEvent.click(submit);

		expect(mockSetUsername).toHaveBeenCalled();
	});
});
