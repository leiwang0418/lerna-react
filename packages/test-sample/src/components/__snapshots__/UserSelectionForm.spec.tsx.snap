// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test renders the basic UserSelectionForm 1`] = `
<div>
  <form>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
            data-shrink="true"
          >
            GitHub用户名
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedStart"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart"
              placeholder="请输入GitHub用户名"
              required=""
              type="text"
              value="leiwang0418"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            查看仓库列表
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
